lib/perl/Genome/Model/Tools/Bwa/RunMem.pm

```diff
@@ -403,7 +403,10 @@ sub _sort_commands {
     my $comp_level = 0;
     my $threads = $self->num_threads;
     my $per_thread_mem = $self->_sort_memory_per_thread_mb;
-    my $tmp_path = File::Spec->catfile($self->temp_dir, "samtools-sort");
+    my $tmp_path = $self->temp_dir;
+
+    $tmp_path = Genome::Sys->create_temp_file_path unless $tmp_path;
```

Comment:
@tmooney, Can we merge this PR for the tool change and make a new one to pass the temp_scratch_directory from the AlignmentResult to the tool?